guides/source/asset_pipeline.md

```diff
@@ -868,218 +779,384 @@ each asset will have a different file name when it is changed. This way you
 ```
 
-To disable the assets cache store:
-
-```ruby
-config.assets.configure do |env|
-  env.cache = ActiveSupport::Cache.lookup_store(:null_store)
-end
-```
+This was crucial with HTTP/1.1, which limited 6-8 simultaneous connections per
+domain. With HTTP/2, browsers fetch multiple files in parallel, making bundling
+less critical for modern applications.
 
-Adding Assets to Your Gems
---------------------------
+#### Compression ❌
 
-Assets can also come from external sources in the form of gems.
+Compression encodes files in a more efficient format to reduce their size
+further when delivered to users. A common technique is [Gzip
+compression](https://en.wikipedia.org/wiki/Gzip).
 
-A good example of this is the `jquery-rails` gem.
-This gem contains an engine class which inherits from `Rails::Engine`.
-By doing this, Rails is informed that the directory for this
-gem may contain assets and the `app/assets`, `lib/assets` and
-`vendor/assets` directories of this engine are added to the search path of
-Sprockets.
+For example, a CSS file that's 200KB may compress to just 50KB when Gzipped.
+Browsers automatically decompress such files upon receipt, saving bandwidth and
+improving speed.
 
-Making Your Library or Gem a Pre-Processor
-------------------------------------------
+Minification is a form of compression that removes unnecessary characters (like
+whitespace, comments, and newlines).
 
-Sprockets uses Processors, Transformers, Compressors, and Exporters to extend
-Sprockets functionality. Have a look at
-[Extending Sprockets](https://github.com/rails/sprockets/blob/main/guides/extending_sprockets.md)
-to learn more. Here we registered a preprocessor to add a comment to the end
-of text/css (`.css`) files.
+Before minification:
 
-```ruby
-module AddComment
-  def self.call(input)
-    { data: input[:data] + "/* Hello From my sprockets extension */" }
-  end
-end
+```javascript
+function add(number1, number2) {
+    // Adds two numbers
+    var sum = number1 + number2;
+    return sum;
+}
 ```
 
-Now that you have a module that modifies the input data, it's time to register
-it as a preprocessor for your MIME type.
+After minification:
 
-```ruby
-Sprockets.register_preprocessor "text/css", AddComment
+```javascript
+function add(a,b){return a+b;}
 ```
 
+Modern ES6+ syntax is supported by modern browsers, thus reducing the reliance
+on minification.
+
+### Sprockets vs. Propshaft
+
+#### Load Order
+
+In Sprockets, you could link files together to ensure they loaded in the correct
+order. For example, a main JavaScript file that depended on other files would
+automatically have its dependencies managed by Sprockets, ensuring everything
+loaded in the right sequence. Propshaft, on the other hand, does not
+automatically handle these dependencies, and instead [lets you manage the asset
+load order manually](#asset-load-order).
+
+#### Versioning
+
+Sprockets simplifies asset fingerprinting by appending a hash to filenames
```

Comment:
```suggestion
Sprockets simplified asset fingerprinting by appending a hash to filenames
```

Reply:
"simplifies" seems like the correct tense. Even though Sprockets is no longer the default, it still simplifies asset fingerprinting in that way.